src/app/core/browse/browse.service.ts

```diff
@@ -103,42 +109,26 @@ export class BrowseService {
           args.push(`page=${options.pagination.currentPage - 1}`);
           args.push(`size=${options.pagination.pageSize}`);
         }
+        if (isNotEmpty(options.startsWith)) {
+          args.push(`startsWith=${options.startsWith}`);
+        }
         if (isNotEmpty(args)) {
           href = new URLCombiner(href, `?${args.join('&')}`).toString();
         }
         return href;
       }),
-      map((endpointURL: string) => new BrowseEntriesRequest(this.requestService.generateRequestId(), endpointURL)),
-      configureRequest(this.requestService)
+      getBrowseEntriesFor(this.requestService, this.rdb)
     );
-
-    const href$ = request$.pipe(map((request: RestRequest) => request.href));
-
-    const requestEntry$ = href$.pipe(getRequestFromRequestHref(this.requestService));
-
-    const payload$ = requestEntry$.pipe(
-      filterSuccessfulResponses(),
-      map((response: GenericSuccessResponse<BrowseEntry[]>) => new PaginatedList(response.pageInfo, response.payload)),
-      map((list: PaginatedList<BrowseEntry>) => Object.assign(list, {
-        page: list.page ? list.page.map((entry: BrowseEntry) => Object.assign(new BrowseEntry(), entry)) : list.page
-      })),
-      distinctUntilChanged()
-    );
-
-    return this.rdb.toRemoteDataObservable(requestEntry$, payload$);
   }
 
   /**
    * Get all items linked to a certain metadata value
-   * @param {string} definitionID     definition ID to define the metadata-field (e.g. author)
    * @param {string} filterValue      metadata value to filter by (e.g. author's name)
-   * @param options                   Options to narrow down your search:
-   *                                  { pagination: PaginationComponentOptions,
-   *                                    sort: SortOptions }
+   * @param options                   Options to narrow down your search
    * @returns {Observable<RemoteData<PaginatedList<Item>>>}
    */
   getBrowseItemsFor(filterValue: string, options: BrowseEntrySearchOptions): Observable<RemoteData<PaginatedList<Item>>> {
-    const request$ = this.getBrowseDefinitions().pipe(
+    return this.getBrowseDefinitions().pipe(
```

Comment:
I'm getting an `IllegalArgumentException` on the rest side because scope isn't defined. Scope has the value "undefined". I think the problem is related with this part of the code:
```
        if (isNotEmpty(options.sort)) {
          args.push(`scope=${options.scope}`);
        }
```

The requested URL: 
`http://localhost/spring-rest/api/discover/browses/dateissued/items?scope=undefined&sort=default,ASC&page=NaN&size=undefined&startsWith=2017`

The logged error on DSpace (java side):
```
2019-02-22 14:07:32,766 WARN  org.dspace.app.rest.utils.ScopeResolver @ The given scope string undefined is not a UUID
java.lang.IllegalArgumentException: Invalid UUID string: undefined
```

Reply:
The issue should be fixed now. I simply forgot to change the "sort" property in the check to "scope". Nice catch!